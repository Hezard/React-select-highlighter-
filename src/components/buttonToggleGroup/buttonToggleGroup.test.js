import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ButtonToggleGroup from './buttonToggleGroup';

describe('Button Toggle Group', () => {
  let container, rerender, getByText, btnFirst, btnSecond;
  const onChange = jest.fn();
  const props = {
    className: 'btn-group-test',
    btnClassName: 'btn-test',
    data: [
      {
        id: 0,
        value: 'Btn 1'
      },
      {
        id: 1,
        value: 'Btn 2'
      }
    ],
    onChange
  };

  beforeEach(() => {
    ({ rerender, container, getByText } = render(<ButtonToggleGroup {...props} />));
    btnFirst = getByText(/Btn 1/i);
    btnSecond = getByText(/Btn 2/i);
  });

  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  test('test initial render', () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  test('test toggle click', () => {
    fireEvent.click(btnFirst);
    expect(onChange.mock.calls[0][0]).toEqual({
      id: 0,
      value: 'Btn 1'
    });
  });

  test('test multiple variant toggle click', () => {
    fireEvent.click(btnSecond);
    fireEvent.click(btnFirst);
    expect(onChange.mock.calls[0][0]).toEqual({
      id: 1,
      value: 'Btn 2'
    });
    expect(onChange.mock.calls[1][0]).toEqual({
      id: 0,
      value: 'Btn 1'
    });
  });

  test('test toggle active state', () => {
    rerender(<ButtonToggleGroup {...props} active={{ id: 1, value: 'Btn 2' }} />);
    expect(btnSecond).toHaveClass('btn--active');
  });

  test('test multiple toggle active state', () => {
    rerender(<ButtonToggleGroup {...props} multiple active={{ 'Btn 1': true, 'Btn 2': true }} />);
    expect(btnFirst).toHaveClass('btn--active');
    expect(btnSecond).toHaveClass('btn--active');
  });
});
