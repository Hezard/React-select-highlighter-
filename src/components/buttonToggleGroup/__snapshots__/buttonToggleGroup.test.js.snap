// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Toggle Group test initial render 1`] = `
<div
  class="btn-toggle-group btn-group-test"
>
  <button
    class="btn btn btn-test btn--md btn--primary"
  >
    Btn 1
  </button>
  <button
    class="btn btn btn-test btn--md btn--primary"
  >
    Btn 2
  </button>
</div>
`;
